src/main/java/com/conveyal/analysis/controllers/RegionalAnalysisController.java

```diff
@@ -220,38 +218,16 @@ private HumanKey getSingleCutoffGrid (
         if (!fileStorage.exists(singleCutoffFileStorageKey)) {
             // An accessibility grid for this particular cutoff has apparently never been extracted from the
-            // regional results file before. Extract one and save it for future reuse. Older regional analyses
-            // did not have arrays allowing multiple cutoffs, percentiles, or destination pointsets. The
-            // filenames of such regional accessibility results will not have a percentile or pointset ID.
-            // First try the newest form of regional results: multi-percentile, multi-destination-grid.
+            // regional results file before. Extract one and save it for future reuse.
             String multiCutoffKey = String.format("%s_%s_P%d.access", regionalAnalysisId, destinationPointSetId, percentile);
             FileStorageKey multiCutoffFileStorageKey = new FileStorageKey(RESULTS, multiCutoffKey);
-            if (!fileStorage.exists(multiCutoffFileStorageKey)) {
-                LOG.warn("Falling back to older file name formats for regional results file: " + multiCutoffKey);
-                // Fall back to second-oldest form: multi-percentile, single destination grid.
-                multiCutoffKey = String.format("%s_P%d.access", regionalAnalysisId, percentile);
-                multiCutoffFileStorageKey = new FileStorageKey(RESULTS, multiCutoffKey);
-                if (fileStorage.exists(multiCutoffFileStorageKey)) {
-                    checkArgument(analysis.destinationPointSetIds.length == 1);
-                } else {
-                    // Fall back on oldest form of results, single-percentile, single-destination-grid.
-                    multiCutoffKey = regionalAnalysisId + ".access";
-                    multiCutoffFileStorageKey = new FileStorageKey(RESULTS, multiCutoffKey);
-                    if (fileStorage.exists(multiCutoffFileStorageKey)) {
-                        checkArgument(analysis.travelTimePercentiles.length == 1);
-                        checkArgument(analysis.destinationPointSetIds.length == 1);
-                    } else {
-                        throw AnalysisServerException.notFound("Cannot find original source regional analysis output.");
-                    }
-                }
-            }
             LOG.debug("Single-cutoff grid {} not found on S3, deriving it from {}.", singleCutoffKey, multiCutoffKey);
 
-            InputStream multiCutoffInputStream = new FileInputStream(fileStorage.getFile(multiCutoffFileStorageKey));
+            InputStream multiCutoffInputStream = FileUtils.getInputStream(fileStorage.getFile(multiCutoffFileStorageKey));
             Grid grid = new SelectingGridReducer(cutoffIndex).compute(multiCutoffInputStream);
 
             File localFile = FileUtils.createScratchFile(fileFormat.toString());
-            FileOutputStream fos = new FileOutputStream(localFile);
+            OutputStream fos = FileUtils.getOutputStream(localFile);
```

Comment:
Looks like some of the methods that take `fos` as a parameter (at least Grid.write) don't use buffered output streams, so using this utlity method that buffers the output should greatly improve write speeds.